### PR TITLE
feat(replays): Add click event search support to API endpoints

### PR DESCRIPTION
**Pull request by cmanallen**
Adds support for querying by select HTML attributes which were clicked on.  The following fields are filterable in the search box:

- `replay_click.alt`
- `replay_click.class`
- `replay_click.id`
- `replay_click.label`
- `replay_click.role`
- `replay_click.tag`
- `replay_click.testid`
- `replay_click.text`
- `replay_click.title`

This has a dependency on https://github.com/getsentry/snuba/pull/3855

### CODE REVIEW

File: src/sentry/replays/post_process.py [hunk 1]
```diff
@@ -70,6 +70,14 @@ def generate_normalized_output(
...
 
         item.pop("isArchived")
+        item.pop("click_tag", None)
+        item.pop("click_id", None)
+        item.pop("click_class", None)
+        item.pop("click_role", None)
+        item.pop("click_testid", None)
+        item.pop("click_aria_label", None)
+        item.pop("click_alt", None)
+        item.pop("click_title", None)
```
Comment (ryan953) [inline src/sentry/replays/post_process.py:80]: sort these please. Makes it easier to compare with the same list in other place

Comment (ryan953) [reply src/sentry/replays/post_process.py:80]: missing `text`?

File: src/sentry/replays/post_process.py [hunk 2]
```diff
@@ -70,6 +70,14 @@ def generate_normalized_output(
         item["urls"] = item.pop("urls_sorted", None)
 
         item.pop("isArchived")
+        item.pop("click_tag", None)
+        item.pop("click_id", None)
+        item.pop("click_class", None)
+        item.pop("click_role", None)
+        item.pop("click_testid", None)
+        item.pop("click_aria_label", None)
```
Comment (ryan953) [inline src/sentry/replays/post_process.py:78]: should this be `aria_label` or just label?

blueprint says we're going to expose `replay_click.label` as the field name. We could be consistent to drop `aria` internally to the code too

Comment (cmanallen) [reply src/sentry/replays/post_process.py]: Corrected this to expose the field as `replay_click.label`.  However, internally we'll continue the aria_label naming.  The value is extracted from that HTML attribute and needs to retain its specificity.

**Review by JoshFerge — APPROVED** [review]

**Comment by cmanallen** [comment]
https://github.com/getsentry/sentry/issues/45524

**Pull request merged**